src/Middleware/DeltaMiddleware.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace SilverStripe\Config\Middleware;
+
+use InvalidArgumentException;
+use SilverStripe\Config\Collections\DeltaConfigCollection;
+use SilverStripe\Config\MergeStrategy\Priority;
+
+/**
+ * Applies a set of user-customised modifications to config
+ */
+class DeltaMiddleware implements Middleware
+{
+    use MiddlewareCommon {
+        __construct as __commonConstruct;
```

Comment:
Double underscore prefixed methods are reserved in PHP (http://php.net/manual/en/language.oop5.magic.php)

Reply:
Perhaps the trait shouldn't even have a `__construct` method and instead should be `boot` or `init`? (https://stackoverflow.com/a/12583603/4978044)

Reply:
I'll look into it thanks.